editors/vscode/src/wasm_preview.ts

```diff
@@ -3,116 +3,49 @@
 
 // cSpell: ignore winit
 
-import { Uri, TextDocumentShowOptions } from "vscode";
+import { Uri } from "vscode";
 import * as vscode from "vscode";
 import { BaseLanguageClient } from "vscode-languageclient";
 
 let previewPanel: vscode.WebviewPanel | null = null;
-let previewUrl: Uri | null = null;
-let previewAccessedFiles = new Set();
-let previewComponent: string = "";
-let queuedPreviewMsg: any | null = null;
-let previewBusy = false;
-let uriMapping = new Map<string, string>();
+let from_lsp_queue: object[] = [];
+let to_lsp_queue: object[] = [];
+
+let language_client: BaseLanguageClient | null = null;
 
 /// Initialize the callback on the client to make the web preview work
 export function initClientForPreview(client: BaseLanguageClient | null) {
-    client?.onRequest("slint/preview_message", async (msg: any) => {
-        if (previewPanel) {
-            // map urls to webview URL
-            if (msg.command === "highlight") {
-                msg.data.path = previewPanel.webview
-                    .asWebviewUri(Uri.parse(msg.data.path, true))
```

Comment:
Where is this webviewUri mapping done now?

Reply:
I see, so the preview don't need the webview url?
Does that means that the LSP also forward the image data?

Reply:
I think wasm cannot just load the image through data, it needs actual URL.  So there might be a need to cummunicate between the wasm preview and vscode to patch these URL